packages/rule-meta-theme-color/src/rule.ts

```diff
@@ -85,34 +88,33 @@ export default class MetaThemeColorRule implements IRule {
             /*
              * `theme-color` can accept any CSS `<color>`:
              *
-             *    * https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color
-             *    * https://drafts.csswg.org/css-color/#typedef-color
-             *
-             *  However:
-             *
-             *    * Values such as `hwb` and `hex with alpha` are not
-             *      supported everywhere `theme-color` is.
+             *   * https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color
+             *   * https://drafts.csswg.org/css-color/#typedef-color
              *
-             *    * `rgba` and `hsla` even if supported, browsers ignore
-             *      the alpha.
-             *
-             *    * Windows/Microsoft Store require the color value to
-             *      be specified either as `hex` or as a color name.
+             *  However, `HWB` and `hex with alpha` values are not
+             *  supported everywhere `theme-color` is. Also, values
+             *  such as `currentcolor` don't make sense, but they will
+             *  be catched by the above check.
              *
              *  Also, some values such as `currentcolor` don't make
              *  sense, but they will be catched by the previous check.
              *
              *  See also:
              *
-             *    * https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#Browser_compatibility
-             *    * https://cs.chromium.org/chromium/src/third_party/WebKit/Source/platform/graphics/Color.cpp?rcl=6263bcf0ec9f112b5f0d84fc059c759302bd8c67
+             *   * https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#Browser_compatibility
+             *   * https://cs.chromium.org/chromium/src/third_party/WebKit/Source/platform/graphics/Color.cpp?rcl=6263bcf0ec9f112b5f0d84fc059c759302bd8c67
              */
 
-            const hexWithoutAlphaRegex = /^#([0-9a-fA-F]{3}){1,2}$/;
-            const colorNameRegex = /^[a-zA-Z]+$/;
+            const hexWithAlphaRegex = /^#([0-9a-fA-F]{4}){1,2}$/;
+
+            if (
+                // `RGBA` support depends on the browser.
+                (color.model === 'rgb' &&
```

Comment:
Move this to a function.

Reply:
Done.

Reply:
I was thinking in move everything between `()` to a function, something like `if (!supported(color, normalizedColorValue))`

what do you think?